### DISCUSSION TITLE
When using PKCE flow in Nextjs, is the `/api/auth/callback` api handler required?

### DISCUSSION THREAD

**Discussion by remorses** [supabase/supabase#14873] [Questions] [answered]
From what i understand PKCE works on client side too without manually calling `exchangeCodeForSession()` so i should be able to redirect the user to a page that uses the supabase client and the session will be stored in the cookies

So the callback api handler is only required if you want to redirect to a page that calls `getSession` on the server, right? because in that case you still need to call `exchangeCodeForSession()` (that can only run automatically on the browser)

Do i understand correctly? If yes it's a good idea to add this information in the Nextjs integration pages

**Comment by J0** [comment, marked as answer]
Thanks for flagging this! This sounds about right - feel free to sound off if there are further issues. Should be in the latest docs as well
